### ISSUE TITLE
Some parallel backend tests failing with NumPy 1.22

### ISSUE THREAD

**Issue by gmarkall** [numba/numba#7760]
For various tests in `numba.tests.test_parallel_backend`, errors are of the form:

```
======================================================================
FAIL: test_threading_concurrent_mix_use_masks_omp (numba.tests.test_parallel_backend.TestSpecificBackend)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/esc/miniconda3/envs/numba-dev/lib/python3.9/site-packages/numba/tests/test_parallel_backend.py", line 396, in test_template
    o, e = self.run_test_in_separate_process(injected_method, backend)
  File "/Users/esc/miniconda3/envs/numba-dev/lib/python3.9/site-packages/numba/tests/test_parallel_backend.py", line 374, in run_test_in_separate_process
    return self.run_cmd(cmdline, env_copy)
  File "/Users/esc/miniconda3/envs/numba-dev/lib/python3.9/site-packages/numba/tests/test_parallel_backend.py", line 362, in run_cmd
    raise AssertionError(
AssertionError: process failed with code 1: stderr follows
OMP: Info #271: omp_set_nested routine deprecated, please use omp_set_max_active_levels instead.
E
======================================================================
ERROR: test_threading_concurrent_mix_use_masks (numba.tests.test_parallel_backend.TestParallelBackend)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/esc/miniconda3/envs/numba-dev/lib/python3.9/site-packages/numba/tests/test_parallel_backend.py", line 336, in test_method
    self.run_compile(impl, parallelism=p)
  File "/Users/esc/miniconda3/envs/numba-dev/lib/python3.9/site-packages/numba/tests/test_parallel_backend.py", line 283, in run_compile
    thread_impl(fnlist)
  File "/Users/esc/miniconda3/envs/numba-dev/lib/python3.9/site-packages/numba/tests/test_parallel_backend.py", line 180, in run_compile
    raise RuntimeError(_msg % '\n'.join([repr(x) for x in errors]))
RuntimeError: Error(s) occurred in delegated runner:
UFuncTypeError(<ufunc 'ufunc_foo'>, (<class 'numpy.dtype[float32]'>, <class 'numpy.dtype[float32]'>, None))
UFuncTypeError(<ufunc 'ufunc_foo'>, (<class 'numpy.dtype[float32]'>, <class 'numpy.dtype[float32]'>, None))
UFuncTypeError(<ufunc 'ufunc_foo'>, (<class 'numpy.dtype[float32]'>, <class 'numpy.dtype[float32]'>, None))
UFuncTypeError(<ufunc 'ufunc_foo'>, (<class 'numpy.dtype[float32]'>, <class 'numpy.dtype[float32]'>, None))

----------------------------------------------------------------------
Ran 1 test in 20.943s
```

**Comment by 1844144** [comment]
Appears in DUFunc, fixed in numpy 1.22.3

**Comment by stuartarchibald** [comment]
@1844144 thanks for checking.
